src/bundle.ts

```diff
@@ -232,19 +232,21 @@ function makeBundleVisitor(version: OasMajorVersion, dereference: boolean, rootD
     ctx: UserContext,
   ) {
     const [fileRef, pointer] = [target.location.source.absoluteRef, target.location.pointer];
+    const componentsGroup = components[componentType];
 
-    const pointerBase = pointerBaseName(pointer);
-    const refBase = refBaseName(fileRef);
+    let name = '';
 
-    let name = pointerBase || refBase;
+    const refParts = pointer.slice(2).split('/').filter(Boolean);  // slice(2) removes "#/"
```

Comment:
What if there is no `pointer`? For example, in the test file `openapi-with-external-refs.yaml` above line 9:

`- $ref: ./param-b.yaml`

Is it the case where `name ? _${name}  : ''` will be an empty string?

Reply:
Yes, exactly.